[executorch] Move llm custom ops to extension

lucylq opened the pull request.

larryliu0820 reviewed (approved):
This looks good, though we need to pay attention to the CI jobs that generates the pip wheel. I think Hansong ran into issues for the release

lucylq commented:
> This looks good, though we need to pay attention to the CI jobs that generates the pip wheel. I think Hansong ran into issues for the release

ah, good callout. Adding ciflow/binaries for wheel build tests

lucylq committed:
Move llm custom ops to extension (#4467)

Summary:
Move sdpa from examples/models/llama2/custom_ops to extension/llm/custom_ops.

Consolidate llm custom ops into one place - later, other custom ops (eg. pad, reshape) can be placed in the same place.

Pull Request resolved: https://github.com/pytorch/executorch/pull/4467

Test Plan:
Imported from GitHub, without a `Test Plan:` line.

Build executorch, export and run stories with xnnpack and quant in oss.

Reviewed By: helunwencser

Differential Revision: D60469989

Pulled By: lucylq
